File: server/src/main/java/com/springboot/store/entity/PurchaseReturn.java
package com.springboot.store.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder

@Entity
@Table(name = "purchase_return")
public class PurchaseReturn {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int subtotal;
    private int discount;
    private int total;
    private String note;
    private Date createdDate;

    @OneToMany(mappedBy = "purchaseReturn", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<PurchaseReturnDetail> purchaseReturnDetails;

    @ManyToOne
    @JoinColumn(name = "purchase_order_id")
    private PurchaseOrder purchaseOrder;

    @ManyToOne
    @JoinColumn(name = "staff_id")
    private Staff staff;

    @ManyToOne
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;

    @ManyToOne
    @JoinColumn(name = "store_id")
    private Store store;
}

File: entities/PurchaseReturn.ts
export type PurchaseReturn = {
    id: number;
    subtotal: number;
    discount: number;
    total: number;
    note: string;
    createdDate: string;
    purchaseReturnDetails: PurchaseReturnDetail[];
    staffId: number;
    supplierId: number;
  };
  
  export type PurchaseReturnDetail = {
    id: number;
    quantity: number;
    supplyPrice: number;
    returnPrice: number;
    note: string;
    productId: number;
  };
  